services/gitlab/gitlab-coverage.service.js

```diff
@@ -0,0 +1,121 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const {
+  coveragePercentage: coveragePercentageColor,
+} = require('../color-formatters')
+const { optionalUrl } = require('../validators')
+const { BaseSvgScrapingService, NotFound, redirector } = require('..')
+
+const badgeSchema = Joi.object({
+  message: Joi.alternatives()
+    .try(
+      Joi.string().regex(/^([0-9]+\.[0-9]+%)|unknown$/),
+      Joi.equal('unknown')
+    )
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  gitlab_url: optionalUrl,
+}).required()
+
+const documentation = `
+<p>
+  Important: If your project is publicly visible, but the badge is like this:
+  <img src="https://img.shields.io/badge/build-not&nbsp;found-red" alt="build not found"/>
+</p>
+<p>
+  Check if your pipelines are publicly visible as well.<br />
+  Navigate to your project settings on GitLab and choose General Pipelines under CI/CD.<br />
+  Then tick the setting Public pipelines.
+</p>
+<p>
+  Now your settings should look like this:
+</p>
+<img src="https://user-images.githubusercontent.com/12065866/67156911-e225a180-f324-11e9-93ad-10aafbb3e69e.png" alt="Setting Public pipelines set"/>
+<p>
+Also make sure you have set up code covrage parsing as described <a href="https://docs.gitlab.com/ee/ci/pipelines/settings.html#test-coverage-parsing">here</a>
+</p>
+<p>
+  Your badge should be working fine now.
+</p>
+`
+
+class GitlabCoverage extends BaseSvgScrapingService {
```

Comment:
Can (and should) export the class this way once the redirector is removed to allow for using the service test helpers

```suggestion
module.exports = class GitlabCoverage extends BaseSvgScrapingService {
```